flowzone.yml

```diff
@@ -2369,6 +2394,41 @@ jobs:
           path: ${{ runner.temp }}/docs.tar.zst
           retention-days: 90
 
+  npm_sbom:
+    name: Generate SBOM for NPM
+    runs-on: ${{ fromJSON(inputs.runs_on) }}
+    needs:
+      - is_npm
+      - npm_test
+      - versioned_source
+    if: ${{ needs.is_npm.outputs.npm == 'true' && needs.is_npm.outputs.npm_sbom == 'true' }}
+    <<: *customWorkingDirectory
+    steps:
+      - *getGitHubAppToken
+      - *checkoutVersionedSha
+      - *createLocalRefs
+      - *setupNode
+
+      - run: npm install
+
+      - name: Generate SBOM
+        run: |
+          npx @cyclonedx/cyclonedx-npm --output-file=${{ runner.temp }}/bom.xml
+
+      - <<: *publishSBOMArtifacts
+        with:
+          name: sbom-npm-${{ github.event.pull_request.head.sha }}
```

Comment:
I don't think there is a need to append the head sha to the artifact name here, it would be much cleaner without it?

Reply:
You would also want `github_publish` to depend on these jobs if you go that route.

Reply:
good call